modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DefaultDLEditFileEntryDisplayContext.java

```diff
@@ -81,6 +84,17 @@ public DefaultDLEditFileEntryDisplayContext(
 			storageEngine);
 	}
 
+	@Override
+	public String getAllMimeTypeSizeLimit() {
+		Map<String, Long> mimeTypeSizeLimit = _dlValidator.getMimeTypeSizeLimit(
+			_dlRequestHelper.getSiteGroupId());
+
+		JSONObject mimeTypeSizeLimitJSONObject =
+			JSONFactoryUtil.createJSONObject(mimeTypeSizeLimit);
+
+		return mimeTypeSizeLimitJSONObject.toString();
```

Comment:
Why do we serialize into jsonObject to parse in front?
Isn't it better to use a `HashMapBuilder.<String, Object>` inside HashMapBuilder.<String, Object> like [here](https://github.com/liferay/liferay-portal/blob/f42cc1f8b56d9413773ad6ad99772bac05ab8ed0/modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/TranslateDisplayContext.java#L246-L253).

Reply:
I updated source to pass the object to front instead of passing a string.